sonar-db/src/main/java/org/sonar/db/purge/ComponentDisabledListener.java

```diff
@@ -17,32 +17,9 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.server.computation.dbcleaner;
+package org.sonar.db.purge;
 
-import java.util.List;
-import org.sonar.api.server.ServerSide;
-import org.sonar.db.purge.PurgeListener;
-import org.sonar.server.issue.index.IssueIndexer;
-import org.sonar.server.test.index.TestIndexer;
-
-@ServerSide
-public class IndexPurgeListener implements PurgeListener {
-  private final TestIndexer testIndexer;
-  private final IssueIndexer issueIndexer;
-
-  public IndexPurgeListener(TestIndexer testIndexer, IssueIndexer issueIndexer) {
-    this.testIndexer = testIndexer;
-    this.issueIndexer = issueIndexer;
-  }
-
-  @Override
-  public void onComponentDisabling(String uuid) {
-    testIndexer.deleteByFile(uuid);
-  }
-
-  @Override
-  public void onIssuesRemoval(String projectUuid, List<String> issueKeys) {
-    issueIndexer.deleteByKeys(projectUuid, issueKeys);
-  }
+public interface ComponentDisabledListener {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Annotate the "ComponentDisabledListener" interface with the @FunctionalInterface annotation [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1609)